docs/source/data/errors.mdx

```diff
@@ -604,21 +604,60 @@ In this case, the error above is reported to Apollo Studio as:
 The REDACTED doesn't have sufficient privileges.
 ```
 
-## Returning HTTP status codes
+## Setting HTTP status code and headers
 
 GraphQL, by design, does not use the same conventions from REST to communicate via HTTP verbs and status codes. Client information should be contained in the schema or as part of the standard response `errors` field. We recommend using the included [Error Codes](#built-in-error-codes) or [Custom Errors](#custom-errors) for error consistency rather than directly modifying the HTTP response.
 
-You can set custom fields on your HTTP response by using a [plugin](/apollo-server/integrations/plugins). Be aware that GraphQL client libraries may not treat all response status codes the same, and so it will be up to your team to decide what patterns to use.
+Apollo Server's general behavior with respect to status codes is as follows:
+- If Apollo Server has not properly started up or is in the process of shutting down, use status code 500. (The former only happens if you're using a serverless integration instead of `await`ing `startStandaloneServer(server)` or `server.start()`; the latter only happens if you have not properly set up draining for your non-standalone integration.)
```

Comment:
I feel like you can remove the second sentence here; I think the first sentence is clear enough that the extra background isn't necessary. 
```suggestion
- If Apollo Server hasn't properly started up or is in the process of shutting down, it responds with a 500 status code.
```

Reply:
My point here is that if you're seeing this it's because you've set up your server in a non-optimal way (which is easy to do because it does take a bit more work in anything other than startStandaloneServer to do it right) — so if people come here concerned their server is sending 500s it would be nice to tell them there's a better way. Maybe could be expressed better though?

Reply:
Okay, I see what you mean. 

Could you further clarify what "if you're using a serverless integration instead of `await`ing `startStandaloneServer(server)` or `server.start()`" means here? Does that mean this error code will only happen with serverless integrations where the server doesn't start up properly? Or if you forget to `await` your server's startup with non-serverless integrations? Or both? 


I think we should we could make this a sub-bullet point, so something like (this suggestion will need to be tweaked based on the answer to my question above):

```suggestion
-  If Apollo Server hasn't properly started up or is in the process of shutting down, it responds with a 500 status code. 
  - The former happens if you haven't `await`ed your server's start-up, and the latter if you aren't properly [draining your server](/apollo-server/api/plugin/drain-http-server/#using-the-plugin).
```

Reply:
Yeah, basically the entire difference as far as AS4 is returned between server-y and serverless integrations is that server-y integrations have good startup error handling (you first try to start the server machinery, and only if it works do you move on to actually listening for HTTP requests) and server-less integrations have crappy startup error handling (there's no way to do async work before accepting requests, so any startup errors just turn into "send a 500 for every request").

The proper pattern for server-y integrations involves asserting when you call a function like `expressMiddleware` that the server is started, so the startup issue should be a serverless integration or a poorly written server-y integration.

Reply:
So this would be something more like this:
```suggestion
-  If Apollo Server hasn't correctly started up or is in the process of shutting down, it responds with a 500 status code. 
  - The former can happen if you use a serverless integration and it sends requests to an Apollo Server instance that hasn't properly started. The latter happens if you aren't properly [draining your server](/apollo-server/api/plugin/drain-http-server/#using-the-plugin).
```

Reply:
Oh the other thing is that serverless will wait to start... so it really means that there was an error. This should be good, I'll commit it:

So this would be something more like this:
```suggestion
-  If Apollo Server hasn't correctly started up or is in the process of shutting down, it responds with a 500 status code. 
  - The former can happen if you use a serverless integration and it sends requests to an Apollo Server instance that had an error on startup. The latter happens if you aren't properly [draining your server](/apollo-server/api/plugin/drain-http-server/#using-the-plugin).
```